example/js/lib/api/diplomat-runtime.js

```diff
@@ -80,14 +80,18 @@ export class DiplomatBuf {
     this.ptr = ptr;
     this.size = size;
     this.free = () => {
-      wasm.diplomat_free(this.ptr, this.size, align);
-      DiplomatBuf_finalizer.unregister(this);
+      const successfully_unregistered = DiplomatBuf_finalizer.unregister(this);
+      if (successfully_unregistered) {
+        wasm.diplomat_free(this.ptr, this.size, align);
+      } else {
+        console.error(`Failed to unregister DiplomatBuf at ${ptr}, this is a bug. Either it was never registered (leak), it was already unregistered (failed attempt to double free), or the unregister token was unrecognized (fallback to GC).`);
```

Comment:
Suggestion: put a similar check into finalization registries that are auto-generated, not just DiplomatBuf

EDIT: Nevermind; this appears to be the only place we use `unregister`. Hopefully having this code here will mean that if we add more `unregister` in the future then this will get copied into place.